src/components/executive-education-2u/ExecutiveEducation2UPage.jsx

```diff
@@ -1,98 +1,82 @@
-/* eslint-disable react/no-danger */
 import React, {
-  useContext, useEffect, useMemo, useState,
+  useContext, useEffect, useMemo,
 } from 'react';
-import { Container, Tabs, Tab } from '@edx/paragon';
+import { Container } from '@edx/paragon';
 import { AppContext } from '@edx/frontend-platform/react';
-import { camelCaseObject } from '@edx/frontend-platform/utils';
 import { Helmet } from 'react-helmet';
 import Skeleton from 'react-loading-skeleton';
 import { logError } from '@edx/frontend-platform/logging';
 
 import NotFoundPage from '../NotFoundPage';
-import { getExecutiveEducation2UTerms } from './data';
+import UserEnrollmentForm from './UserEnrollmentForm';
+import {
+  useActiveQueryParams,
+  useExecutiveEducation2UContentMetadata,
+} from './data';
 
 function ExecutiveEducation2UPage() {
   const { enterpriseConfig } = useContext(AppContext);
-  const [isLoading, setIsLoading] = useState(true);
-  const [terms, setTerms] = useState();
+  const activeQueryParams = useActiveQueryParams();
+
+  const isExecEd2UFulfillmentEnabled = useMemo(() => (
+    enterpriseConfig.enableExecutiveEducation2UFulfillment && activeQueryParams.has('course_uuid')
+  ), [enterpriseConfig, activeQueryParams]);
+
+  const {
+    isLoading: isLoadingContentMetadata,
+    contentMetadata,
+  } = useExecutiveEducation2UContentMetadata({
+    courseUUID: activeQueryParams.get('course_uuid'),
+    isExecEd2UFulfillmentEnabled,
+  });
 
   useEffect(() => {
-    const fetchTerms = async () => {
-      setIsLoading(true);
-      try {
-        const response = await getExecutiveEducation2UTerms();
-        setTerms(camelCaseObject(response.data));
-      } catch (error) {
-        logError(error);
-      } finally {
-        setIsLoading(false);
-      }
-    };
-    if (enterpriseConfig.enableExecutiveEducation2UFulfillment) {
-      fetchTerms();
-    }
-  }, [enterpriseConfig.enableExecutiveEducation2UFulfillment]);
+    const isExecEd2UEnabled = enterpriseConfig.enableExecutiveEducation2UFulfillment;
```

Comment:
Couldn't reuse that name here because it's already defined in the upper scope. However, having these are separate variables isn't really necessary, so I'm just using `enterpriseConfig.enableExecutiveEducation2UFulfillment` directly in the conditional statement now, instead.